lib/utils/authentication.js

```diff
@@ -44,23 +45,114 @@ module.exports.register = function(server, options, next) {
      * Even though this is a synchronous function, it returns a promise
      * This will allow for an easy transition to async key generators
      * @param {string} username - unencrypted username
-     * @return {Bluebird} resolves to a credentials object
+     * @return {Promise} resolves to a credentials object
      *   where credentials is an object with the following properties:
-     *   `id, key, algorithm, issueTime, expireTime`
+     *   `id, key, algorithm, issueTime, expireTime, studyRoles`
      */
     authUtils.generateHawkCredentials = function(username) {
-        const credentials = {
-            username: username,
-            id: uuid.v4(),
-            key: uuid.v4(),
-            algorithm: config.algorithm,
-            issueTime: +new Date(),
-            expireTime: +new Date() + config.hawkKeyLifespan
-        };
-        return Bluebird.resolve(credentials);
+
+        return refreshCurrentStudyRoles(username)
+            .then((roleList) => {
+                const credentials = {
+                    username: username,
+                    id: uuid.v4(),
+                    key: uuid.v4(),
+                    algorithm: config.algorithm,
+                    issueTime: + new Date(),
+                    expireTime: + new Date() + config.hawkKeyLifespan,
+                    studyRoles: roleList
+                };
+
+                return credentials;
+            });
     };
 
     /**
+     * Retrieves the current roles for a user, revokes old roles, and
+     * adds the new roles
+     * @param {string} username - the username to retrieve roles
+     * @return {object} roles - the current roles
+     */
+    function refreshCurrentStudyRoles(username) {
+        const encryptedUsername = authUtils.encryptUsername(username);
```

Comment:
I would agree, however I think that that should be reserved for model and collection usage(maybe?). Here we're just encrypting a string, and I feel that grabbing a model and using it just for decryption feels a bit couply. I think this will make more sense if/once we transition to using just one decryption key, to have a function that de/encrypts all data. It feels a bit weird to have a Model basically alias this function just for usernames.